lib/hanami/cli/commands/app/generate/action.rb

```diff
@@ -35,16 +35,16 @@ def initialize(fs: Dry::Files.new, inflector: Dry::Inflector.new,
             end
 
             # rubocop:disable Metrics/ParameterLists
-            def call(slice:, name:, url: nil, http: nil, format: DEFAULT_FORMAT, skip_view: DEFAULT_SKIP_VIEW, **)
-              slice = inflector.underscore(Shellwords.shellescape(slice))
+            def call(name:, url: nil, http: nil, format: DEFAULT_FORMAT, skip_view: DEFAULT_SKIP_VIEW, slice: nil, **)
+              slice = inflector.underscore(Shellwords.shellescape(slice)) unless slice.nil?
               name = inflector.underscore(Shellwords.shellescape(name))
               *controller, action = name.split(ACTION_SEPARATOR)
 
               if controller.empty?
                 raise ArgumentError.new("cannot parse controller and action name: `#{name}'\n\texample: users.show")
               end
 
-              generator.call(slice, controller, action, url, http, format, skip_view)
+              generator.call(app.namespace, controller, action, url, http, format, skip_view, slice)
```

Comment:
I wonder whether we really need the "controller" terminology here at all. I know this is mostly internal code but AFAICT we'll only ever refer to these as plain "actions" everywhere else.

This is really just an (optionally nested) action path, e.g. "foo", or "foo.bar.baz". Is there any reason we couldn't just pass that through as a singular string and then work out the nested directory aspect over in the generator?

IMO this would make the generator easier to understand, because I didn't really know what the "controller" arg meant when I was just looking at it now. I had to flip back up to here to figure it out.

(I'd be happy for this to be a follow up refactor, in the spirit of not wanting to block this PR!)

Reply:
Let's cleanup later then. I'm going on vacation in less than 3hrs 😎 🌴 .